dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/trackedentity/AbstractTrackedEntityInstanceService.java

```diff
@@ -1147,7 +1137,7 @@ private TrackedEntityInstance getTei( org.hisp.dhis.trackedentity.TrackedEntityI
             return null;
         }
 
-        Set<TrackedEntityAttribute> readableAttributesCopy = new HashSet<>( readableAttributes );
+        Set<TrackedEntityAttribute> readableAttributesCopy = readableAttributes;
```

Comment:
I think you can keep the old code here, so readableAttributeCopy is indeed a cope from the start. Right now it's the same reference. See next comment for more.